### PR TITLE
[DD 422] CTA Popup

### PR DESCRIPTION
**Pull request by teesloane**
# Description

[Ticket](https://circleci.atlassian.net/browse/DD-422)
[Preview Link](http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/DD-422-popup-preview/?force-all)

This PR adds a popup CTA when an unregistered user has seen/been to three pages on our docs.

https://user-images.githubusercontent.com/12987958/158244945-0a8a793b-54e0-426c-a5ca-906847a8e3eb.mp4

## Notes

- Refactored out cta button to a mixin, and re-implemented it in the first-steps doc.

### CODE REVIEW

File: src/js/experiments/popupCta.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  if (!timesVisited || isNaN(timesVisited)) {
+    timesVisited = 1;
+    localStorage.setItem(LS_KEY, timesVisited);
+  }
+  return timesVisited;
+}
+
+function incrementTimesVisited() {
+  const timesVisited = getTimesVisited();
+  if (timesVisited == SHOW_POPUP_AFTER_N_TIMES) {
```
Comment (chantelle11) [inline src/js/experiments/popupCta.js:34]: nit: use === ?

Comment (chantelle11) [inline src/js/experiments/popupCta.js:34]:
```suggestion
  if (timesVisited === SHOW_POPUP_AFTER_N_TIMES) {
```

File: src/styles/_popup.scss [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  width: 100%;
+  height: 100vh;
+  background-color: rgba(0, 0, 0, 0.8);
+  z-index: 1031;
+  top: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  left: 0;
+  display: none;
```
Comment (chantelle11) [inline src/styles/_popup.scss:14]: This has display: flex and display: none inside

**Review by chantelle11 — COMMENTED** [review]
I love this! Just a few questions
1. Can we not show the banner if they are already logged in?
2. Is there an option to 'not show me this again today'? Or can we implicitly just not show it to them again until next day? or for at least an hour? Show popup after 3 clicks seems to much to me

**Review by chantelle11 — COMMENTED** [review]
There seems to be an image missing top right corner
Screen Shot 2022-03-14 at 9 43 40 PM

**Review by rsylvian — APPROVED** [review]
LGTM ✅ nice work!

You can merge but please leave the traffic to 0% so we can make sure to do a round of prod testing before we release it to the world 🙂

File: jekyll/_includes/popup_cta.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+       class="logo"
+       alt="circleci"/>
+  <h2 class="heading">6,000 Minutes Free</h2>
+  <h3 class="sub-heading">CircleCI, now with more build minutes <br> than anyone else for FREE 🎉</h3>
+  <a class="popup-signup-cta" target="_blank" href="https://circleci.com/signup/">Sign up for Free</a>
+
+  <div class="teams-callout-wrapper">
+    <h3 class="teams-callout">These teams get it</h3>
+    <div class="popup-bottom-row">
+        <img class="popup-teams-bubbles" crossorigin="anonymous" src="{{ site.baseurl }}/assets/img/docs/cci_customer_teams.png">
```
Comment (rsylvian) [inline jekyll/_includes/popup_cta.html:17]: can we add an `alt` please?

File: jekyll/_includes/popup_cta.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        <button class="popup-skip-btn">
+        <span class="skip-text">Skip</span>
+        <img class="arrow" crossorigin="anonymous" src="{{ site.baseurl }}/assets/img/docs/arrow-right.svg">
```
Comment (rsylvian) [inline jekyll/_includes/popup_cta.html:20]: same here

File: src/js/experiments/popupCta.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  if (popupData.timesVisited === SHOW_POPUP_AFTER_N_TIMES) {
+    popupData.timesVisited = 1;
+    popupData.lastSeen = new Date();
+  } else {
+    popupData.timesVisited += 1;
+  }
+  localStorage.setItem(LS_KEY, JSON.stringify(popupData));
+}
+
+// https://app.optimizely.com/v2/projects/16812830475/experiments/21253750305/
```
Comment (rsylvian) [inline src/js/experiments/popupCta.js:72]: This is the wrong link I believe

**Pull request merged**